fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java

```diff
@@ -2332,6 +2332,10 @@ private PlanFragment visitPhysicalJoin(PlanFragment leftFragment, PlanFragment r
                 throw new StarRocksPlannerException("unknown join operator: " + node, INTERNAL_ERROR);
             }
 
+            if (optExpr.getOutputProperty().getDistributionProperty().isAny()) {
```

Comment:
Some physical rewrite rule will rebuild the optExpression and it may miss setting the outProperty field like `AddDecodeNodeForDictStringRule`.  I think you could add a `optExpr.getOutputProperty() != null` here for simple. I will cp #32496 to main and add this field to optExpressionBuilder and set it.

Reply:
OK

Reply:
fixed. thanks.